editor/modules/map/editor-state/index.ts

```diff
@@ -154,6 +153,7 @@ export class EditorState implements MouseHandler, KeyHandler {
         switch (this.state) {
             case State.Palette: next_state = this.palette_area.mouseMovePalette(event); break;
             case State.Editor: next_state = this.editor_area.mouseMoveEditor(event); break;
+            default: throw 'Unreachable code reached';
```

Comment:
Oh I see what you mean. Here it's not possible because the compiler is enforcing the fact that `this.state` can't be null because its type is `State` which is an `enum`. So trying to assign `null` or `undefined` to `this.state` is a compile error.